WordPress/WordPressUITests/LoginTests.swift

```diff
@@ -25,86 +23,25 @@ class LoginTests: XCTestCase {
-        usernameEmailTextField.tap()
-        usernameEmailTextField.typeText(WordPressTestCredentials.oneStepUser)
-
-        let passwordSecureTextField = app.secureTextFields["Password"]
-        passwordSecureTextField.tap()
-        passwordSecureTextField.typeText(WordPressTestCredentials.oneStepPassword)
+        simpleLogin(username: WordPressTestCredentials.oneStepUser, password: WordPressTestCredentials.oneStepPassword)
 
-        app.buttons["Log In"].tap()
-
-        self.waitForElementToAppear(app.tabBars["Main Navigation"])
+        waitForElementToAppear(element: app.tabBars[ elementStringIDs.mainNavigationBar ])
     }
 
+    func testUnsuccessfulLogin() {
+        simpleLogin(username: WordPressTestCredentials.oneStepUser, password: "password")
 
-    func testTwoStepLogin() {
-        let usernameEmailTextField =  app.textFields["Username / Email"]
-        usernameEmailTextField.tap()
-        usernameEmailTextField.typeText(WordPressTestCredentials.twoStepUser)
-
-        let passwordSecureTextField = app.secureTextFields["Password"]
-        passwordSecureTextField.tap()
-        passwordSecureTextField.typeText(WordPressTestCredentials.twoStepPassword)
-
-        app.buttons["Log In"].tap()
-
-        self.waitForElementToAppear(app.tabBars["Main Navigation"])
+        waitForElementToAppear(element: app.images[ "icon-alert" ])
+        app.buttons.element(boundBy: 1).tap()
     }
 
     func testSelfHostedLoginWithoutJetPack() {
-        app.buttons["Add Self-Hosted Site"].tap()
-
-        let usernameEmailTextField =  app.textFields["Username / Email"]
-        usernameEmailTextField.tap()
-        usernameEmailTextField.typeText(WordPressTestCredentials.selfHostedUser)
-
-        let passwordSecureTextField = app.secureTextFields["Password"]
-        passwordSecureTextField.tap()
-        passwordSecureTextField.typeText(WordPressTestCredentials.selfHostedPassword)
-
-        let siteURLTextField = app.textFields["Site Address (URL)"]
-        siteURLTextField.tap()
-        siteURLTextField.typeText(WordPressTestCredentials.selfHostedSiteURL)
-
-        app.buttons["Add Site"].tap()
-
-        self.waitForElementToAppear(app.tabBars["Main Navigation"])
-
-        app.tabBars["Main Navigation"].buttons["My Sites"].tap()
-
-        app.tabBars["Main Navigation"].buttons["My Sites"].tap()
-
-        let cellName = WordPressTestCredentials.selfHostedSiteName
-        app.tables.cells.staticTexts[cellName].tap()
-
-        app.tables.elementBoundByIndex(0).swipeUp()
-
-        app.tables.cells.staticTexts["Settings"].tap()
-
-        app.tables.elementBoundByIndex(0).swipeUp()
+        loginSelfHosted(username: WordPressTestCredentials.selfHostedUser, password: WordPressTestCredentials.selfHostedPassword, url: WordPressTestCredentials.selfHostedSiteURL)
 
-        app.tables.cells.staticTexts["Remove Site"].tap()
+        waitForElementToAppear(element: app.tabBars[ elementStringIDs.mainNavigationBar ])
```

Comment:
For what it's worth, on my slow internet (I'm currently in rural Canada) I had to bump the timeout to 10s here for this test to ever pass. May be worth considering, since this test really does touch the internet :P